mlflow/server/js/src/components/MetricsPlotPanel.js

```diff
@@ -126,8 +130,38 @@ export class MetricsPlotPanel extends React.Component {
 
   handleLineSmoothChange = (lineSmoothness) => this.setState({ lineSmoothness });
 
+  updatePopover = (data) => {
+    this.isClicked = !this.isClicked;
+
+    // Ignore double click.
+    setTimeout(() => {
+      if (this.isClicked) {
+        this.isClicked = false;
+        const popover = this.popoverRef.current;
+        const { visible, x, y } = popover.state;
+        const { points, event: { clientX, clientY } } = data;
+        const clickedSamePoint = x === clientX && y === clientY;
+
+        const runItems = points.map((point) => ({
+          name: point.data.name,
+          color: point.fullData.marker.color,
+          runUuid: point.data.runUuid,
+        })).reverse();
+
+        // Instead of setting state on MetricPlotPanel (current component),
+        // should call a method on the Popover directly via the ref.
+        popover.setState({
+          visible: !visible || !clickedSamePoint,
+          x: clientX,
+          y: clientY,
+          runItems,
+        });
```

Comment:
I'd recommend creating a method in RunLinksPopover with 4 parameters to avoid exposing child components state directly to the parent component.

Reply:
Do you mean to create a method that looks like this?

```javascript
class RunLinksPopover extends React.Component {
  ...

  method = (visible, x, y, runItems) => this.setState(...);

  ...
}
```

Reply:
Yes.